crucible/src/Lang/Crucible/Backend.hs

```diff
@@ -613,6 +626,59 @@ readPartExpr bak (PE p v) msg = do
   addAssertion bak (LabeledPred p (SimError loc msg))
   return v
 
+
+-- | Run the CHC solver on the current proof obligations, and return the
+-- solution as a substitution from the uninterpreted functions to their
+-- definitions.
+runCHC ::
+  (IsSymBackend sym bak, sym ~ ExprBuilder t st fs, MonadIO m) =>
+  bak ->
+  [SomeSymFn sym] ->
+  m (MapF (SymFnWrapper sym) (SymFnWrapper sym))
+runCHC bak uninterp_inv_fns  = liftIO $ do
+  let sym = backendGetSym bak
+
+  implications <- proofObligationsAsImplications bak
+  clearProofObligations bak
+
+  -- log to stdout
+  let logData = defaultLogData
+        { logCallbackVerbose = \_ -> putStrLn
+        , logReason = "SAW inv"
```

Comment:
This code lives in Crucible, not SAW... shouldn't the `logReason` mention Crucible instead?